models/asymkey/ssh_key.go

```diff
@@ -429,9 +430,8 @@ func SynchronizePublicKeys(usr *user_model.User, s *auth.Source, sshPublicKeys [
 			newKeys = append(newKeys, key)
 		}
 	}
-	if AddPublicKeysBySource(usr, s, newKeys) {
-		sshKeysNeedUpdate = true
-	}
+
+	addedKeys := AddPublicKeysBySource(usr, s, newKeys)
```

Comment:
That's probably
```suggestion
	addedKeys = AddPublicKeysBySource(usr, s, newKeys)
```
after above.